pennylane_cirq/cirq_device.py

```diff
@@ -167,58 +159,56 @@ def operations(self):
+    def apply(self, operations, rotations=None, **kwargs):
+        rotations = rotations or []
 
-        # This code is adapted from the pennylane-qiskit plugin
-        for e in self.obs_queue:
-            # Identity and PauliZ need no changes
-            if e.name == "PauliX":
-                # X = H.Z.H
-                self.apply("Hadamard", wires=e.wires, par=[])
+        for i, operation in enumerate(operations):
+            if operation.name == "BasisState":
+                if i > 0:
+                    raise qml.DeviceError(
+                        "The operation BasisState is only supported at the beginning of a circuit."
+                    )
 
-            elif e.name == "PauliY":
-                # Y = (HS^)^.Z.(HS^) and S^=SZ
-                self.apply("PauliZ", wires=e.wires, par=[])
-                self.apply("S", wires=e.wires, par=[])
-                self.apply("Hadamard", wires=e.wires, par=[])
+                self.apply_basis_state(operation)
+            elif operation.name == "QubitStateVector":
+                if i > 0:
+                    raise qml.DeviceError(
+                        "The operation QubitStateVector is only supported at the beginning of a circuit."
+                    )
 
-            elif e.name == "Hadamard":
-                # H = Ry(-pi/4)^.Z.Ry(-pi/4)
-                self.apply("RY", e.wires, [-np.pi / 4])
+                self.apply_qubit_state_vector(operation)
+            else:
+                cirq_operation = self._complete_operation_map[operation.name]
 
-            elif e.name == "Hermitian":
-                # For arbitrary Hermitian matrix H, let U be the unitary matrix
-                # that diagonalises it, and w_i be the eigenvalues.
-                Hmat = e.parameters[0]
-                Hkey = tuple(Hmat.flatten().tolist())
+                # If command is None do nothing
+                if cirq_operation:
+                    cirq_operation.parametrize(*operation.parameters)
 
-                if Hmat.shape not in [(2, 2), (4, 4)]:
-                    raise qml.DeviceError(
-                        "Cirq only supports single-qubit and two-qubit unitary gates and thus only single-qubit and two-qubit Hermitian observables."
+                    self.circuit.append(
+                        cirq_operation.apply(
+                            *[self.qubits[wire] for wire in operation.wires]
+                        )
                     )
 
-                if Hkey in self._eigs:
-                    # retrieve eigenvectors
-                    U = self._eigs[Hkey]["eigvec"]
-                else:
-                    # store the eigenvalues corresponding to H
-                    # in a dictionary, so that they do not need to
-                    # be calculated later
-                    w, U = np.linalg.eigh(Hmat)
-                    self._eigs[Hkey] = {"eigval": w, "eigvec": U}
+        # TODO: get pre rotated state here
```

Comment:
?

Reply:
Actually I'm a little confused with the handling of pre-rotation state and rotated state. In the implementation of `default.qubit`, the probability is given for the rotated state but the state is given for the pre_rotated_state. I think this is somewhat inconsistent, isn't it?

Here in Cirq I have the additional problem that I can't really extract the pre_rotated_state without much hassle

Reply:
What is your opinion on this, @antalszava ?

Reply:
And @josh146

Reply:
This is slightly intentional. Now that there is `qml.prob`, we no longer want the user to access `dev.probability`, so for consistency this should always represent the actual device probability.

For state, however, we are still in an awkward position where there is no equivalent to extract it from the qnode yet

Reply:
So what do you think of me keeping the post-rotation state as `dev.state`? It is possible to extract the pre-rotation state, but it will not make the code clearer...

Reply:
I think it is fine for now. `dev.state` should only ever be used for debugging, and we can attach caveats to it that depend on the device (e.g. a lot of hardware simulators don't even store any state)

Reply:
I added a docstring that explains this